src/components/account/login.vue

```diff
@@ -69,10 +79,63 @@ export default {
       password: ''
     };
   },
+  computed: {
+    oidcLoginPath() {
+      const { query } = this.$route;
+      const next = typeof query.next === 'string' ? query.next : null;
+      const qs = queryString({ next });
+      return `/v1/oidc/login${qs}`;
+    }
+  },
+  created() {
+    this.handleOIDCError();
+  },
   mounted() {
-    this.$refs.email.focus();
+    if (this.config.oidcEnabled)
+      window.addEventListener('pageshow', this.reenableIfPersisted);
+    else
+      this.$refs.email.focus();
+  },
+  beforeUnmount() {
+    if (this.config.oidcEnabled)
+      window.removeEventListener('pageshow', this.reenableIfPersisted);
   },
   methods: {
+    verifyNewSession() {
+      const sessionExpires = localStore.getItem('sessionExpires');
+      const newSession = sessionExpires == null ||
+        Number.parseInt(sessionExpires, 10) < Date.now();
+      if (!newSession) this.alert.info(this.$t('alert.alreadyLoggedIn'));
```

Comment:
how about navigating to home page instead of show alert?

Reply:
We thought about doing that in the regular/non-OIDC case, but we thought it'd be confusing if the credentials entered didn't match the user who was already logged in, or if the credentials were incorrect (because they wouldn't be checked, yet it'd seem like they were). It's definitely a confusing user experience though, and I've filed getodk/central#400 to improve it.

That said, your comment is making me realize that we're able to do something different in the OIDC case. There are no credentials entered in that case, so nothing to be mismatched or incorrect, and I think what you describe would be helpful behavior.

Reply:
this is not a blocker, I feel this can happen in next release or later

Reply:
Sounds good. I probably won't get to it for .4, but I've linked to this discussion from getodk/central#400. It'd be great to make that change at some point.